codepropertygraph/src/main/scala/io/shiftleft/passes/DiffGraphProtoSerializer.scala

```diff
@@ -59,11 +61,13 @@ class DiffGraphProtoSerializer {
   def serialize(diffGraph: DiffGraph): DiffGraphProto = {
     import DiffGraph.Change._
     val builder = DiffGraphProto.newBuilder
+
     def newEntry = DiffGraphProto.Entry.newBuilder
+
     diffGraph.iterator
       .map {
         case SetNodeProperty(node, key, value) =>
-          newEntry.setNodeProperty(addNodeProperty(node.id2, key, value))
+          newEntry.setNodeProperty(addNodeProperty(node.id2, key, value, null))
```

Comment:
If the appliedDiffGraph is only sometimes passed, please pass it as an `Option` rather than handling `null` around. 
This makes the code more explicit ("it may not be set") and avoids spurious NPE that may be hard to debug.